pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -106,39 +116,28 @@ public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusM
                             null, statusDate, null,
                             rs.getLong(DBConstants.TISSUE_ID), rs.getLong(DBConstants.DSM_KIT_REQUEST_ID), null));
                 } else {
-                    dbVals.resultException = new RuntimeException(
-                            "Couldn't get ddp instance id for order " + mercuryStatusMessage.getOrderID());
+                    dbVals.resultException = new DsmInternalError(String.format("Couldn't get ddp instance id for order %s",
+                            mercuryStatusMessage.getOrderID()));
                 }
             } catch (SQLException ex) {
                 dbVals.resultException =
-                        new RuntimeException("Error getting ddp instance id for order " + mercuryStatusMessage.getOrderID(), ex);
+                        new DsmInternalError(String.format("Error getting ddp instance id for order %s", mercuryStatusMessage.getOrderID()), ex);
             }
             return dbVals;
         });
         if (results.resultException != null) {
-            throw new DSMPubSubException("Unable to process the status of the order " + baseMercuryStatusMessage, results.resultException);
+            throw new DsmInternalError("Unable to process the status of the order " + baseMercuryStatusMessage, results.resultException);
```

Comment:
There is no need to stack the exceptions here. Simply throwing the existing exception is sufficient.

Reply:
If we don't stack it, it will just be treated as `Exception` in the functions that call it, so it needs to get added to the method signature. Is that really what we want?

Reply:
I understand now, I think given how it has been in the process of debugging errors on either side of Mercury and DSM it is going to be beneficial for us to log `baseMercuryStatusMessage` ( it does not contain PHI, just some barcode values and the error that happened ). We can log it to see what they sent back. Especially since sometimes they apparently send a huge string that we don't want to keep all of it, it's good to have a sense of what was happening.

Reply:
I fixed L104 and pushed again, I didn't change the stacking before because I thought the main reason for the stacking looking unnecessary to you was the `baseMercuryStatusMessage`  in there, which I think we should log. So now I moved the `baseMercuryStatusMessage` to when the original exception is created and removed the stacking